### PR TITLE
Adding callback to version-check task and moving the version-check fr…

### PR DESCRIPTION
**Pull request by nikhilmodak**
…om dependencies of plugin-init to plugin-upload

### CODE REVIEW

**Comment by meghameshram** [comment]
What about the versioncheck.js changes?

File: lib/plugin-create.js [hunk 1]
```diff
@@ -59,15 +59,25 @@ module.exports = function (gulp, gutil) {
         'res/editors/tinymce/**/*.js',
         'res/editors/tinymce/instances/**/*.json',
         'res/editors/tinymce/plugins/**/img/*.{png,gif,jpg,jpeg,svg}',
-        'res/feature/*/*/res/editors/tinymce/**/*.js',
```
Comment (nikhilmodak) [inline lib/plugin-create.js]: Checkin in this file by mistake in this PR - but we will need them anyways

File: gulp/plugin.js [hunk 2]
```diff
@@ -129,11 +129,16 @@ module.exports = function (gulp, gutil) {
     return stream;
   });
 
-  gulp.task('plugin', [
-    gutil.env['skip-upload'] ?
-      'plugin-ready' :
-      'plugin-upload'
-  ]);
+  var pluginTaskDependencies = [];
+  if (gutil.env['skip-version-check']) {
```
Comment (dougschroeder) [inline gulp/plugin.js]: shouldn't this be `if (!gutil.env['skip-version-check']) {` ?

Comment (nikhilmodak) [inline gulp/plugin.js]: yes you are right, my bad

Comment (nikhilmodak) [inline gulp/plugin.js]: https://github.com/lithiumtech/lithium-sdk/commit/491b84990581bd1bf8ae6ac3e1a8d180cf121a3e

**Pull request merged**